packages/state-history/src/lib/entities-state-history.ts

```diff
@@ -0,0 +1,404 @@
+import {
+  Store,
+  OrArray,
+  coerceArray,
+  isUndefined,
+  filterNil,
+  StoreValue,
+} from '@ngneat/elf';
+import { Subscription, pairwise, filter, map } from 'rxjs';
+import {
+  EntitiesState,
+  getIdType,
+  EntitiesRef,
+  getEntityType,
+  DefaultEntitiesRef,
+  upsertEntities,
+  getEntitiesIds,
+  selectEntity,
+  selectEntities,
+} from '@ngneat/elf-entities';
+
+export interface EntitiesStateHistoryOptions<E extends EntitiesRef> {
+  maxAge: number;
```

Comment:
where is the comparator?

Reply:
I guess I misunderstood. I thought I always need to check only by refs. Elf under the hood is checking whether entity changed or not(`select` operator).

Reply:
So am I just bringing this back `comparatorFn: () => true`?

Reply:
Yes, thanks

You can finalize it afterwards.

Reply:
So now `comparatorFn` is for extra checking, e.g. deep equal but still, it has ref checking under the hood in the `select` operator

Reply:
Let's skip it for now

Reply:
I left it and add tests and description. Let it be as an extra check